pkg/cloud/cloud.go

```diff
@@ -393,7 +393,7 @@ func (c *cloud) CreateDisk(ctx context.Context, volumeName string, diskOptions *
 	if iops > 0 {
 		requestInput.Iops = aws.Int64(iops)
 	}
-	if throughput > 0 && diskOptions.VolumeType == VolumeTypeGP3 {
+	if throughput > 0 && createType == VolumeTypeGP3 {
```

Comment:
Is this check `createType == VolumeTypeGP3` necessary? throughput will only ever be > 0 if `createType = VolumeTypeGP3` or `createType == ""`: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/dd719c12ef9cbf8e2b86152003b3afb8fa0057e9/pkg/cloud/cloud.go#L332

Reply:
That's not true, we don't check the volume type when pass through the throughput in controller.go